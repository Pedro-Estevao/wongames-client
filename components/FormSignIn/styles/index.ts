'use client';

import styled, { css } from 'styled-components';
import { lighten } from 'polished';
import Link from "next/link";

export const ForgotPassword = styled(Link)`
	${({ theme }) => css`
		display: block;
		font-size: ${theme.font.sizes.small};
		color: ${theme.colors.black};
		text-decoration: none;
		text-align: right;

		&:hover {
			color: ${lighten(0.2, theme.colors.black)};
		}
	`}
`;
