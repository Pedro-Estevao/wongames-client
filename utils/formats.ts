export const formatPrice = (price: number | bigint) => {
	return new Intl.NumberFormat('en', {
		style: 'currency',
		currency: 'USD'
	}).format(price);
};

export const formatDate = (date: string) => {
	return new Intl.DateTimeFormat('en-US', {
		day: 'numeric',
		month: 'short',
		year: 'numeric'
	}).format(new Date(date));
};

export const formatImageUrl = (url: string | undefined) => {
	if (process.env.NEXT_PUBLIC_API_URL) {
		return `${process.env.NEXT_PUBLIC_API_URL}${url}`;
	}

	if (url) {
		return url;
	}

	return null;
};