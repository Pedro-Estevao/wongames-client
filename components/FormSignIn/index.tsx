'use client';

import React from 'react';
import Link from 'next/link';
import * as S from './styles';
import Button from '../Button';
import TextField from '../TextField';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEnvelope, faLock } from '@fortawesome/free-solid-svg-icons';

const FormSignIn = () => {
	return (
		<S.FormSignInContainer>
			<form>
				<TextField
					name="email"
					placeholder="Email"
					type="email"
					icon={<FontAwesomeIcon icon={faEnvelope} />}
				/>
				<TextField
					name="password"
					placeholder="Password"
					type="password"
					icon={<FontAwesomeIcon icon={faLock} />}
				/>
				<S.ForgotPassword href="#">Forgot your password?</S.ForgotPassword>

				<Button size="large" fullWidth>
					Sign in now
				</Button>

				<S.FormLink>
					Donâ€™t have an account? <Link href="/sign-up">Sign up</Link>
				</S.FormLink>
			</form>
		</S.FormSignInContainer>
	);
};

export default FormSignIn;
