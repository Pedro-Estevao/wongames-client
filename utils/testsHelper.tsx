import { CartContext, CartContextData, CartContextDefaultValues } from "@/hooks/useCart";
import { WishlistContext, WishlistContextData, WishlistContextDefaultValues } from "@/hooks/useWishlist";
import { theme } from "@/public/styles/theme";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

type CustomRenderProps = {
    cartProviderProps?: CartContextData;
    wishlistProviderProps?: WishlistContextData;
} & Omit<RenderOptions, 'queries'>;

const customRender = (
    ui: React.ReactElement,
    {
        cartProviderProps = CartContextDefaultValues,
        wishlistProviderProps = WishlistContextDefaultValues,
        ...renderOptions
    }: CustomRenderProps = {}
) => render(
    <ThemeProvider theme={theme}>
        <CartContext.Provider value={cartProviderProps}>
            <WishlistContext.Provider value={wishlistProviderProps}>
                {ui}
            </WishlistContext.Provider>
        </CartContext.Provider>
    </ThemeProvider>,
    renderOptions
);

export * from "@testing-library/react";
export { customRender as render };