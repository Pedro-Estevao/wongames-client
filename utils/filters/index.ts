import { ItemProps } from '@/components/ExploreSidebar';
import { FilterObject, FilterValue } from "@/components/Layouts/Games";
import { ParsedUrlQueryInput } from 'querystring';

type ParseArgs = {
	queryString: ParsedUrlQueryInput;
	filterItems: Pick<ItemProps, 'type' | 'name'>[];
};

export const parseQueryStringToFilter = ({
	queryString,
	filterItems
}: ParseArgs) => {
	const obj: Record<string, string | string[]> = {};

	Object.keys(queryString).forEach((key) => {
		const item = filterItems?.find((item) => item.name === key);
		const isCheckbox = item?.type === 'checkbox';
		const queryValue = queryString[key];

		if (Array.isArray(queryValue)) {
			// Remove duplicatas do array
			obj[key] = Array.from(new Set(queryValue));
		} else if (typeof queryValue === 'string') {
			if (isCheckbox) {
				// Se for um checkbox, converte a string em array e remove duplicatas
				const values = queryValue.split(',').filter(Boolean); // Remove strings vazias
				obj[key] = Array.from(new Set(values));
			} else {
				// Se não for checkbox, mantém a string como está
				obj[key] = queryValue;
			}
		}
	});

	return obj;
};

export const parseQueryStringToObjFilter = (
	queryString: Record<string, string | string[]>,
	filterItems: ItemProps[]
): FilterObject => {
	const filters: FilterObject = {};

	filterItems.filter((item) => item.name !== 'sort').forEach((item) => {
		if (queryString[item.name]) {
			const queryValue = queryString[item.name];
			let filterValue: FilterValue | string | string[] | number;

			if (Array.isArray(queryValue)) {
				filterValue = queryValue;
			} else {
				// Check if the value is a number (used for price)
				filterValue = isNaN(Number(queryValue)) ? queryValue : Number(queryValue);

				if (item.type === 'checkbox' && typeof filterValue === 'string') {
					filterValue = filterValue.split(',').map(str => str.trim());
				}
			}

			filters[item.name] = item.value
				? item.filter
					? { [item.value]: { [item.filter]: filterValue } }
					: { [item.value]: filterValue }
				: item.filter
					? { [item.filter]: filterValue }
					: filterValue;
		}
	});

	return filters;
};