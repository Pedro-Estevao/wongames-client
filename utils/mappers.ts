import { BannerProps } from '@/components/Banner';
import { GameCardProps } from '@/components/GameCard';
import { HighlightProps } from '@/components/Highlight';
import {
	BannerEntity,
	BannerEntityResponseCollection,
	BannerFragmentFragment,
	GameEntityResponseCollection,
	GameFragmentFragment,
	HighlightFragmentFragment
} from '@/graphql/generated/graphql';

export const bannerMapper = (banners: BannerEntityResponseCollection) => {
	return (
		(banners?.data?.map((banner: BannerEntity) => ({
			img: `${process.env.NEXT_PUBLIC_API_URL}${(
				banner.attributes as BannerFragmentFragment
			)?.image?.data?.attributes?.url}`,
			title: (banner.attributes as BannerFragmentFragment)?.title || '',
			subTitle: (banner.attributes as BannerFragmentFragment)?.subTitle || '',
			buttonLabel:
				(banner.attributes as BannerFragmentFragment)?.button?.label || '',
			buttonLink:
				(banner.attributes as BannerFragmentFragment)?.button?.link || '',
			...((banner.attributes as BannerFragmentFragment)?.ribbon && {
				ribbon:
					(banner.attributes as BannerFragmentFragment)?.ribbon?.text || null,
				ribbonColor:
					(banner.attributes as BannerFragmentFragment)?.ribbon?.color || null,
				ribbonSize:
					(banner.attributes as BannerFragmentFragment)?.ribbon?.size || null
			})
		})) as BannerProps[]) || []
	);
};

export const gamesMapper = (games: GameEntityResponseCollection) => {
	return (
		(games?.data.map((game) => ({
			title: (game.attributes as GameFragmentFragment).name,
			slug: (game.attributes as GameFragmentFragment).slug,
			developer: (game.attributes as GameFragmentFragment).developers?.data[0]
				.attributes?.name,
			img: `${process.env.NEXT_PUBLIC_API_URL}${(game.attributes as GameFragmentFragment)
				.cover?.data?.attributes?.url}`,
			price: (game.attributes as GameFragmentFragment).price
		})) as GameCardProps[]) || []
	);
};

export const staticGamesParams = (games: GameEntityResponseCollection) => {
	return (
		(games?.data.map((game) => ({
			title: (game.attributes as GameFragmentFragment).name,
			slug: (game.attributes as GameFragmentFragment).slug,
			developer: (game.attributes as GameFragmentFragment).developers?.data[0]
				.attributes?.name,
			img: `${process.env.NEXT_PUBLIC_API_URL}${(game.attributes as GameFragmentFragment)
				.cover?.data?.attributes?.url}`,
			price: (game.attributes as GameFragmentFragment).price
		})) as GameCardProps[]) || []
	);
};

export const highlightMapper = (highlight: HighlightFragmentFragment) => {
	return (
		highlight &&
		({
			title: highlight.title,
			subTitle: highlight.subTitle,
			backgroundImage: `${process.env.NEXT_PUBLIC_API_URL}${highlight.background?.data?.attributes?.url}`,
			floatImage: `${process.env.NEXT_PUBLIC_API_URL}${highlight.floatImage?.data?.attributes?.url}`,
			buttonLabel: highlight.buttonLabel,
			buttonLink: highlight.buttonLink,
			alignment: highlight.alignment
		} as HighlightProps)
	);
};
