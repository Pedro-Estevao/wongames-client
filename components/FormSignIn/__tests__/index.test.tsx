import { render, screen } from '@/utils/testsHelper';
import { expect } from '@jest/globals';
import FormSignIn from "@/components/FormSignIn";
import { useRouter } from "next/navigation";

jest.mock('next/navigation', () => ({
    useRouter: jest.fn(),
    useSearchParams: jest.fn().mockReturnValue(new URLSearchParams('callbackUrl=/dashboard')),
}));

const push = jest.fn();

jest.mock('next-auth', () => ({
    useSession: jest.fn(),
    signIn: jest.fn(),
    signOut: jest.fn(),
}));

beforeEach(() => {
    (useRouter as jest.Mock).mockImplementation(() => ({
        push,
        query: '',
        asPath: '',
        route: '/'
    }));
});

describe('<FormSignIn />', () => {
    it('should render the form', () => {
        const { container } = render(<FormSignIn />);

        expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument();
        expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument();
        expect(
            screen.getByRole('button', { name: /sign in now/i })
        ).toBeInTheDocument();

        expect(container.parentElement).toMatchSnapshot();
    });

    it('should render the forgot password link', () => {
        render(<FormSignIn />);

        expect(
            screen.getByRole('link', { name: /forgot your password\?/i })
        ).toBeInTheDocument();
    });

    it('should render text to sign up if doesn’t have an account', () => {
        render(<FormSignIn />);

        expect(screen.getByRole('link', { name: /sign up/i })).toBeInTheDocument();
        expect(screen.getByText(/don’t have an account\?/i)).toBeInTheDocument();
    });
});