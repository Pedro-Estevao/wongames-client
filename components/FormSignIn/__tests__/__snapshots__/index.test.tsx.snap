// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignIn /> should render the form 1`] = `
.c5 {
  display: block;
  font-size: 1.4rem;
  color: #030517;
  text-decoration: none;
  text-align: right;
}

.c5:hover {
  color: #0f1971;
}

.c7 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c7:focus {
  box-shadow: 0 0 0 3px #3CD3C1;
}

.c7:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

.c2 {
  display: flex;
  background-color: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c2:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c4 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c4:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEAEA inset;
  filter: none;
}

.c4:-webkit-autofill::first-line {
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
}

.c3 {
  display: flex;
  align-items: center;
  width: 2.2rem;
  color: #8F8F8F;
  order: 0;
}

.c3>svg {
  width: 2.2rem;
  height: 100%;
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c6 {
  margin: 3.2rem auto 1.6rem;
}

.c8 {
  font-size: 1.4rem;
  color: #030517;
  text-align: center;
}

.c8 a {
  color: #3CD3C1;
  text-decoration: none;
  border-bottom: 0.1rem solid #3CD3C1;
  transition: color,border,0.1s ease-in-out;
}

.c8 a:hover {
  color: #29b3a3;
  border-bottom: 0.1rem solid #29b3a3;
}

<body>
  <div>
    <div
      class="c0"
    >
      <form
        method="POST"
      >
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-envelope "
                data-icon="envelope"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              class="c4"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-lock "
                data-icon="lock"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M144 144l0 48 160 0 0-48c0-44.2-35.8-80-80-80s-80 35.8-80 80zM80 192l0-48C80 64.5 144.5 0 224 0s144 64.5 144 144l0 48 16 0c35.3 0 64 28.7 64 64l0 192c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64L0 256c0-35.3 28.7-64 64-64l16 0z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <input
              class="c4"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <a
          class="c5"
          href="/forgot-password"
        >
          Forgot your password?
        </a>
        <button
          class="c6 c7"
          type="submit"
        >
          <span>
            <span>
              Sign in now
            </span>
          </span>
        </button>
        <div
          class="c8"
        >
          Donâ€™t have an account? 
          <a
            href="/sign-up"
          >
            Sign up
          </a>
        </div>
      </form>
    </div>
  </div>
</body>
`;
